### PR TITLE
feat: Moved enrollments charts to new API endpoint.

### PR DESCRIPTION
**Pull request by saleem-latif**
__Jira Ticket:__ [ENT-9420](https://2u-internal.atlassian.net/browse/ENT-9420)

__Description:__
Updated frontend logic according to the backend changes made in https://github.com/openedx/edx-enterprise-data/pull/489 

# For all changes

- [ ] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/AdvanceAnalyticsV2/tabs/AnalyticsTable.jsx [hunk 1]
```diff
@@ -29,6 +32,12 @@ const AnalyticsTable = ({
     endDate,
     // pages index from 1 in backend, frontend components index from 0
     currentPage: currentPage + 1,
+    pageSize,
+  });
+
+  const CSVDownloadURL = EnterpriseDataApiService.getEnrollmentsCSVDownloadURL(enterpriseId, {
+    start_date: startDate,
+    end_date: endDate,
```
Comment (muhammad-ammar) [inline src/components/AdvanceAnalyticsV2/tabs/AnalyticsTable.jsx:40]: `AnalyticsTables` is a common component being used for all data tables in V2. But here we are fetching data for enrollments. I think the changes should be generic to cater all data tables.

File: src/components/AdvanceAnalyticsV2/data/utils.js [hunk 2]
```diff
@@ -8,3 +15,91 @@ const simulateURL = (activeTab, chartType) => {
...
+  if (calculation === CALCULATION.MOVING_AVERAGE_3_PERIODS) {
+    return calculateMovingAverage(timeSeriesData, countKey, 3);
+  }
+  if (calculation === CALCULATION.MOVING_AVERAGE_7_PERIODS) {
+    return calculateMovingAverage(timeSeriesData, countKey, 7);
+  }
+
+  // If no calculation is applied,
+  return timeSeriesData;
+};
```
Comment (muhammad-ammar) [inline src/components/AdvanceAnalyticsV2/data/utils.js:105]: 1. Can we convert the utility method docs to jsdoc?
2. Can we add more details in docs to explain what exactly each util does?

File: src/components/AdvanceAnalyticsV2/data/hooks/useEnterpriseEnrollmentsData.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import { useQuery } from '@tanstack/react-query';
+import _ from 'lodash';
+
+import { useMemo } from 'react';
+import { ANALYTICS_TABS, generateKey } from '../constants';
+import { applyGranularity, applyCalculation } from '../utils';
+import EnterpriseDataApiService from '../../../../data/services/EnterpriseDataApiService';
+
+const applyDataTransformations = (response, granularity, calculation, allowedEnrollTypes = ['certificate', 'audit']) => {
```
Comment (muhammad-ammar) [inline src/components/AdvanceAnalyticsV2/data/hooks/useEnterpriseEnrollmentsData.js:9]: Please add documentation to explain what exactly we are doing?

File: src/components/AdvanceAnalyticsV2/data/hooks/useEnterpriseEnrollmentsData.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  const modifiedResponse = _.cloneDeep(response);
+  if (modifiedResponse && modifiedResponse.data && modifiedResponse.data.enrollmentsOverTime) {
```
Comment (jajjibhai008) [inline src/components/AdvanceAnalyticsV2/data/hooks/useEnterpriseEnrollmentsData.js:11]: [nit] We can simplify this check by using optional chaining `if (modifiedResponse?.data?.enrollmentsOverTime)
`

File: src/data/services/EnterpriseDataApiService.js [hunk 5]
```diff
@@ -204,6 +204,14 @@ class EnterpriseDataApiService {
     const url = `${EnterpriseDataApiService.enterpriseAdminAnalyticsV2BaseUrl}${enterpriseId}/${chartUrl}?${new URLSearchParams(options).toString()}`;
     return EnterpriseDataApiService.apiClient().get(url);
   }
+
+  static getEnrollmentsCSVDownloadURL(enterpriseId, options) {
```
Comment (jajjibhai008) [inline src/data/services/EnterpriseDataApiService.js:208]: Would you please explain the rational behind this service function?

**Review by jajjibhai008 — APPROVED** [review]

File: docs/decisions/0008-application_state_differenciation.rst [hunk 6]
```diff
@@ -33,7 +33,7 @@ We will take a more intentional approach when implementing API data fetching usi
...
-#. Implement a hook when calling the ``useQuery`` method to avoid duplicate code and easy maintainability. See this `hook <https://github.com/openedx/frontend-app-admin-portal/blob/c67c5e4d8a0328fe75cb9d46791a8b733fad8257/src/components/EnterpriseSubsidiesContext/data/hooks.js#L108-L120>`_ as boilerplate as an example, there are `others to reference in the repo <https://github.com/search?q=repo%3Aopenedx%2Ffrontend-app-admin-portal+%2FuseQuery%5C%28%2F&type=code>`_ as additional guides
+#. Implement a hook when calling the ``useQuery`` method to avoid duplicate code and easy maintainability. See this `hook <https://github.com/openedx/frontend-app-admin-portal/blob/c67c5e4d8a0328fe75cb9d46791a8b733fad8257/src/components/EnterpriseSubsidiesContext/data/index.js#L108-L120>`_ as boilerplate as an example, there are `others to reference in the repo <https://github.com/search?q=repo%3Aopenedx%2Ffrontend-app-admin-portal+%2FuseQuery%5C%28%2F&type=code>`_ as additional guides
```
[the remark this answers is not in the document]
Comment (saleem-latif) [reply docs/decisions/0008-application_state_differenciation.rst:36]: I think Pycharm made this change as part of an unrelated refactor, I will revert this change.

File: package.json [hunk 7]
```diff
@@ -41,13 +41,14 @@
     "@tanstack/react-query": "4.36.1",
     "@tanstack/react-query-devtools": "4.36.1",
     "algoliasearch": "4.8.3",
+    "axios": "^1.7.7",
```
[the remark this answers is not in the document]
Comment (saleem-latif) [reply package.json:44]: I was getting an error of missing `axios` while it being used in the code. So, I added this to fix that.

File: src/data/services/EnterpriseDataApiService.js (hunk 5 above)
Comment (saleem-latif) [reply src/data/services/EnterpriseDataApiService.js:208]: Getting the download URL is tied to backend service as it depends on the deployment URL of the service. So, I placed the utility function in here.

**Review by muhammad-ammar — APPROVED** [review]

**Pull request merged**